admin/server/usergroups.go

```diff
@@ -487,22 +487,41 @@ func (s *Server) AddUsergroupMemberUser(ctx context.Context, req *adminv1.AddUse
 		return nil, status.Error(codes.InvalidArgument, "cannot add member to all-users group")
 	}
 
+	claims := auth.GetClaims(ctx)
+	if !claims.OrganizationPermissions(ctx, org.ID).ManageOrgMembers {
+		return nil, status.Error(codes.PermissionDenied, "not allowed to add user group members")
+	}
+
 	user, err := s.admin.DB.FindUserByEmail(ctx, req.Email)
 	if err != nil {
-		return nil, status.Error(codes.InvalidArgument, err.Error())
+		if !errors.Is(err, database.ErrNotFound) {
+			return nil, status.Error(codes.Internal, err.Error())
+		}
+		// did not find user, check if there is any pending invite
+		invite, err := s.admin.DB.FindOrganizationInvite(ctx, org.ID, req.Email)
+		if err != nil {
+			if !errors.Is(err, database.ErrNotFound) {
+				return nil, status.Error(codes.Internal, err.Error())
+			}
+			// there is no pending invite return error
+			return nil, status.Error(codes.FailedPrecondition, "user is not a member of the organization")
+		}
+		// add group to the invite
+		invite.UsergroupIDs = append(invite.UsergroupIDs, group.ID)
```

Comment:
What if `group.ID` is already in `invite.UsergroupIDs`?

Reply:
oh yeah thought of that but forgot to dedupe